Provide getColorDepth() shim for fake TTYs

jagoda opened the pull request:
The AssertionError constructor in Node 10 depends on the TTY
interface. More specifically, if isTTY is true on stdout then
getColorDepth() must also be available. Ava provides a shim for
isTTY in the test workers but does not provide a getColorDepth()
implementation. This leads to unexpected failures when code under
test is using Node's built-in assert module.

<!--

Read the [contributing guidelines](https://github.com/avajs/ava/blob/master/contributing.md). We are excited about pull requests, but please try to limit the scope, provide a general description of the changes, and remember, it's up to you to convince us to land it. If this fixes an open issue, link to it in the following way: `Fixes #321`. New features and bug fixes should come with tests.

-->

sindresorhus commented on lib/worker/fake-tty.js:
```diff
@@ -4,6 +4,9 @@ const options = require('./options').get();
 	Object.assign(process.stdout, {isTTY: true}, options.tty);
+	process.stdout.getColorDepth = function () {
+		return options.tty.colorDepth;
+	};
```
```js
process.stdout.getColorDepth = () => options.tty.colorDepth;
```

jagoda commented:
@sindresorhus I made the change you requested. The build failures seem unrelated to my changes and rather seem to have been introduced in 8663028de2d7a7be88c4a101dafb873c5af87937. Please correct me if I'm wrong and I will happily fix.

sindresorhus commented:
> The build failures seem unrelated to my changes and rather seem to have been introduced in 8663028. Please correct me if I'm wrong and I will happily fix.

Yup. Already fixed.

novemberborn commented on lib/worker/fake-tty.js:
```diff
@@ -4,6 +4,7 @@ const options = require('./options').get();
 
 if (options.tty) {
 	Object.assign(process.stdout, {isTTY: true}, options.tty);
+	process.stdout.getColorDepth = () => options.tty.colorDepth;
```
We should only shim this in Node.js versions where it's expected to be available.

novemberborn commented on lib/fork.js:
```diff
@@ -39,6 +39,7 @@ module.exports = (file, opts, execArgv) => {
 		file,
 		baseDir: process.cwd(),
 		tty: process.stdout.isTTY ? {
+			colorDepth: process.stdout.getColorDepth ? process.stdout.getColorDepth() : 1,
```
Perhaps leave as `undefined` if there is no `getColorDepth()`, and use that as a signal of whether to shim or not?

Or `null` if you prefer.

novemberborn commented on test/integration/custom-assertions.js:
```diff
@@ -0,0 +1,25 @@
+// The AssertionError constructor in Node 10 depends on the TTY interface
+test('node assertion failures are reported to the console when running in a terminal', t => {
+	const options = {
+		dirname: 'fixture/custom-assertions',
```
Could you rename the fixtures (and this integration test) to `node-assertions`? "Custom" implies support for a wider range of assertions, like Chai.

jagoda commented:
@novemberborn thanks for the feedback! I believe that I have addressed all of your concerns. I also added some extra testing around the different possible TTY states. Please let me know if you have any further concerns.

novemberborn commented:
Fantastic work, thanks @jagoda!
